scylla-cql/src/frame/response/mod.rs

```diff
@@ -10,13 +10,43 @@ use std::sync::Arc;
 pub use error::Error;
 pub use supported::Supported;
 
-use crate::errors::CqlResponseKind;
 use crate::frame::protocol_features::ProtocolFeatures;
 use crate::frame::response::result::ResultMetadata;
 use crate::frame::TryFromPrimitiveError;
 
 use super::frame_errors::CqlResponseParseError;
 
+/// Possible CQL responses received from the server
+#[derive(Debug, Copy, Clone)]
+#[non_exhaustive]
+pub enum CqlResponseKind {
```

Comment:
Is this (and `CqlRequestKind`) going to remain `pub`? Does it have to?

Reply:
Yes, `CqlRequestKind` needs to be pub because of the `ConnectionSetupRequestError::request_kind` field. Same for `CqlResponseKind` and e.g. `ConnectionSetupRequestErrorKind::UnexpectedResponse` variant.

Reply:
Nvm, actually `ConnectionSetupRequestError::request_kind` is private. Should we make fields of this error type public?

Reply:
I think that Display is enough public API for users in this case.

Reply:
`CqlRequestKind` and `CqlResponseKind` are public in scylla-cql. They are re-exported as `pub(crate)` in scylla, though.